### PR TITLE
Add .NET Core 3.0 target to Caliburn.Micro.Platform

### PR DESCRIPTION
**Pull request by mjuen**
As #609 seems to be stuck at the moment I took the liberty to replicate the changes and include the review suggestions from @nigel-sampson.

I also renamed the folder containing the shared files from `net46` to `net46-netcore`.

### CODE REVIEW

File: src/Caliburn.Micro.Platform/Caliburn.Micro.Platform.csproj [hunk 1]
```diff
@@ -74,7 +80,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <None Remove="Platforms\net46\AssemblyInfo.cs" />
+    <None Remove="Platforms\net46-netcore\AssemblyInfo.cs" />
```
Comment (nigel-sampson) [inline src/Caliburn.Micro.Platform/Caliburn.Micro.Platform.csproj]: Not related to this PR, but I suspect the whole `ItemGroup` can be removed. VS tends to screw up projects when removing files that are wildcard included.

File: src/Caliburn.Micro.Platform/ActionMessage.cs [hunk 2]
```diff
@@ -191,7 +191,7 @@ void ElementUnloaded(object sender, RoutedEventArgs e)
             }
             else currentElement = context.View;
 
-#if NET
+#if NET || NETCOREAPP
```
Comment (nigel-sampson) [inline src/Caliburn.Micro.Platform/ActionMessage.cs]: Where is `NETCOREAPP` being defined? Would expect to see something in [`Directory.Build.targets`](https://github.com/Caliburn-Micro/Caliburn.Micro/blob/master/src/Directory.Build.targets).

**Comment by nigel-sampson** [comment]
Looking really good, thanks. Need to sort out why PR builds aren't happening to test this out.

File: src/Caliburn.Micro.Platform/ActionMessage.cs (hunk 2 above)
Comment (mjuen) [reply src/Caliburn.Micro.Platform/ActionMessage.cs]: `NETCOREAPP` is a predefined symbol for .NET Core as target framework.
https://docs.microsoft.com/en-us/dotnet/standard/frameworks

Sorry, I did not check the `Directory.Build.targets` file. I have replaced the `NETCOREAPP` symbol with `NETCORE`, which is now defined in [Directory.Build.targets](https://github.com/Caliburn-Micro/Caliburn.Micro/blob/df87adaac6208f07ea8fd727e5c2a2eed75512de/src/Directory.Build.targets). (df87adaac6208f07ea8fd727e5c2a2eed75512de)

File: src/Caliburn.Micro.Platform/Caliburn.Micro.Platform.csproj (hunk 1 above)
Comment (mjuen) [reply src/Caliburn.Micro.Platform/Caliburn.Micro.Platform.csproj]: Removed the ItemGroup in df87adaac6208f07ea8fd727e5c2a2eed75512de

**Comment by nigel-sampson** [comment]
I've enabled PR builds on forks so any further changes should be built for you.

**Comment by mjuen** [comment]
> 
> 
> I've enabled PR builds on forks so any further changes should be built for you (in theory).

Could you change the SDK to one that supports .NET Core 3.0?

Build is currently failing with `The current .NET SDK does not support targeting .NET Core 3.0.  Either target .NET Core 2.2 or lower, or use a version of the .NET SDK that supports .NET Core 3.0`.

**Comment by nigel-sampson** [comment]
Having some issues updating the pipeline. 
> ##[error]C:\Users\VssAdministrator\.nuget\packages\msbuild.sdk.extras\2.0.31\Sdk\Sdk.props(16,5): Error MSB4184: The expression "[System.IO.Path]::GetDirectoryName('')" cannot be evaluated. The path is not of a legal form.

Will work through it some more tonight.

**Comment by nigel-sampson** [comment]
Glad we managed to get there together

**Comment by nigel-sampson** [comment]
Thanks for the hard work @mjuen

**Comment by mjuen** [comment]
No problem :)

It is truly nerve wracking when you assume that the pipeline now works after fixing several errors and then the next one appears.

**Comment by nigel-sampson** [comment]
Hate having to fix pipelines, leaves a lot of terrible commit history.

**Pull request merged**